tests/vmi_preset_lifecycle_test.go

```diff
@@ -0,0 +1,171 @@
+			}
+		}
+
+		assertCreatedPreset := func() func() {
+			return func() {
+				By("Creating preset via oc command")
+				out, err := runOcCreateCommand(presetJsonFile)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				message := fmt.Sprintf("virtualmachineinstancepreset.kubevirt.io \"%s\" created\n", vmPreset.Name)
+				Expect(out).To(Equal(message))
+			}
+		}
+
+		assertDeletedPreset := func() func() {
+			return func() {
+				By("Deleting preset via oc command")
+				out, err := runOcDeletePresetCommand(vmPreset.Name)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				message := fmt.Sprintf("virtualmachineinstancepreset.kubevirt.io \"%s\" deleted\n", vmPreset.Name)
+				ExpectWithOffset(1, out).To(Equal(message))
+			}
+		}
+
+		assertGetPreset := func() func() {
+			return func() {
+				By("Checking preset exists via oc command")
+				EventuallyWithOffset(1, func() bool {
+					out, err := runOcGetCommand("virtualmachineinstancepreset.kubevirt.io")
+					ExpectWithOffset(1, err).ToNot(HaveOccurred())
+					return strings.Contains(out, vmPreset.Name)
+				}, time.Duration(45)*time.Second).Should(BeTrue(), "Timed out waiting for preset to appear")
+			}
+		}
+
+		assertRemovedFile := func(file string) func() {
+			return func() {
+				if _, err := os.Stat(file); !os.IsNotExist(err) {
+					err := os.Remove(file)
+					ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				}
+				ExpectWithOffset(1, file).NotTo(BeAnExistingFile())
+			}
+		}
+
+		testGivenPreset := func() {
+
+			Context("Preset testing", func() {
+
+				It("should succeed to generate a Preset", assertGeneratedPresetJson())
+
+				It("should succeed to create preset using oc command", assertCreatedPreset())
+
+				It("should succeed to get preset using oc command", assertGetPreset())
+
+				It("should succeed to delete the preset using oc command", assertDeletedPreset())
+			})
+		}
+
+		BeforeEach(func() {
+			presetJsonFile = fmt.Sprintf("%s.json", vmPreset.Name)
+			Expect(presetJsonFile).NotTo(BeAnExistingFile())
+		})
+
+		JustBeforeEach(func() {
+			var err error
+			presetJsonFile, err = generatePresetJson(vmPreset)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(presetJsonFile).To(BeAnExistingFile())
+		})
+
+		AfterEach(func() {
+			assertRemovedFile(presetJsonFile)()
+		})
+
+		Context("preset life-cycle testing", func() {
+
+			testGivenPreset()
+		})
+	})
+})
+
+func generatePresetJson(preset *v1.VirtualMachineInstancePreset) (string, error) {
+	data, err := json.Marshal(preset)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate json for preset %s", preset.Name)
+	}
+
+	jsonFile := fmt.Sprintf("%s.json", preset.Name)
+	err = ioutil.WriteFile(jsonFile, data, 0644)
+	if err != nil {
+		return "", fmt.Errorf("failed to write json file %s", jsonFile)
+	}
+	return jsonFile, nil
+}
+
+func runOcCreateCommand(presetJsonFile string) (string, error) {
+	out, err := tests.RunOcCommand("create", "-f", presetJsonFile)
+	return out, err
+}
```

Comment:
if we think these kind of funcs are useful (as oppose to just using RunOcCommand in the test itself, we should move them to the utils, so everybody will be able to use them

Reply:
agree, as kubevirt implement a new cmd RunCommand, i will update all these func